### PR TITLE
Refs #22620 - make UI honor enabled repo types

### PR DESCRIPTION
**Pull request by jlsherrill**
This adds more awareness around what repo types are
enabled, on the lifecycle env page, the main navigation,
and updates detection on the new repo page

### CODE REVIEW

**Comment by jlsherrill** [comment]
requires https://github.com/Katello/bastion/pull/223

**Comment by akofink** [comment]
Getting the same error locally as the CI:

```
NameError: uninitialized constant Katello::RepositoryTypes
```

Is this PR WIP?

**Comment by jlsherrill** [comment]
@akofink sorry, had a mistake, updated.

**Comment by akofink** [comment]
Hrm... now, when trying to create a new repository, the 'Type' dropdown list is empty. There are no JS errors.

**Comment by jlsherrill** [comment]
@akofink sorry about that, updated.

**Comment by jlsherrill** [comment]
@akofink please retest this.  This did a couple of things:

Fixed the global nav, it wasn't hiding nav items based on enabled repo types
adding controlling of enabled content types via a configuration (katello.yml)  see updates to the example

File: config/katello.yaml.example [hunk 1]
```diff
@@ -1,4 +1,12 @@
 :katello:
+  :content_types:
+    :yum: true
+    :file: true
+    :deb: true
+    :puppet: true
+    :docker: true
+    :ostree: true
```
Comment (akofink) [inline config/katello.yaml.example]: I think you should put some comments. Namely that the default is that all registered content types are allowed. Optionally, that the Katello repository types are located and registered from `/lib/katello/repository_types/*.rb`.

**Comment by jlsherrill** [comment]
@akofink added

File: config/katello.yaml.example (hunk 1 above)
Comment (akofink) [reply config/katello.yaml.example]: I mostly wanted a comment saying that the default is enabled: true if content_types array is empty/nonexistant

**Pull request merged**